pkg/api/licenses.go

```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, res)
 }
+
+// ExportLicenses gives users all licenses as a json file.
+//
+//	@Summary		Export all licenses as a json file
+//	@Description	Export all licenses as a json file
+//	@Id				ExportLicenses
+//	@Tags			Licenses
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{array}		models.LicenseJson
+//	@Failure		500	{object}	models.LicenseError	"Failed to fetch obligations"
+//	@Security		ApiKeyAuth
+//	@Router			/licenses/export [get]
+func ExportLicenses(c *gin.Context) {
+
+	var licenses []models.LicenseDB
+	var licenseJson []models.LicenseJson
+	query := db.DB.Model(&models.LicenseDB{})
+	err := query.Find(&licenses).Error
+	if err != nil {
+		er := models.LicenseError{
+			Status:    http.StatusBadRequest,
+			Message:   "Licenses not found",
+			Error:     err.Error(),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusBadRequest, er)
+		return
+	}
+	for _, license := range licenses {
+		jsonLicense := models.LicenseJson{
+			Shortname:       license.Shortname,
+			Fullname:        license.Fullname,
+			Text:            license.Text,
+			Url:             license.Url,
+			AddDate:         license.AddDate.Format(time.RFC3339), // Convert time.Time to string
+			Copyleft:        fmt.Sprintf("%t", license.Copyleft),
+			FSFfree:         fmt.Sprintf("%t", license.FSFfree),
+			OSIapproved:     fmt.Sprintf("%t", license.OSIapproved),
+			GPLv2compatible: fmt.Sprintf("%t", license.GPLv2compatible),
+			GPLv3compatible: fmt.Sprintf("%t", license.GPLv3compatible),
+			Notes:           license.Notes,
+			Fedora:          license.Fedora,
+			TextUpdatable:   fmt.Sprintf("%t", license.TextUpdatable),
+			DetectorType:    license.DetectorType,
+			Active:          fmt.Sprintf("%t", license.Active),
+			Source:          license.Source,
+			SpdxCompatible:  license.SpdxId, // Corrected field name to match your LicenseDB struct
+			Risk:            fmt.Sprintf("%d", license.Risk),
+			Flag:            fmt.Sprintf("%d", license.Flag),
+			Marydone:        fmt.Sprintf("%t", license.Marydone),
+		}
+		licenseJson = append(licenseJson, jsonLicense)
```

Comment:
Not sure but won't doing so won't make `licenseJson` as a nested list?

Reply:
Do we want something like?

```json
{
    "licenses": [
       {...},
       {...}
     ]
}
```

or 

```json
[
    {...},
    {...}
]
```

the above will create the second one.